ticketbuyer.go

```diff
@@ -0,0 +1,233 @@
+	DcrdCert         string `long:"dcrdcert" description:"File containing the dcrd certificate file"`
+	DcrwUser         string `long:"dcrwuser" description:"Wallet RPC user name"`
+	DcrwPass         string `long:"dcrwpass" description:"Wallet RPC password"`
+	DcrwServ         string `long:"dcrwserv" description:"Hostname/IP and port of dcrwallet RPC server to connect to (default localhost:9110, testnet: localhost:19110, simnet: localhost:19557)"`
+	DcrwCert         string `long:"dcrwcert" description:"File containing the dcrwallet certificate file"`
+	DisableClientTLS bool   `long:"noclienttls" description:"Disable TLS for the RPC client -- NOTE: This is only allowed if the RPC client is connecting to localhost"`
+
+	// Deprecated options which have aliases or counterparts in dcrwallet
+	AccountName       string  `long:"accountname" description:"Name of the account to buy tickets from (default: default)"`
+	TicketAddress     string  `long:"ticketaddress" description:"Address to give ticket voting rights to"`
+	PoolAddress       string  `long:"pooladdress" description:"Address to give pool fees rights to"`
+	PoolFees          float64 `long:"poolfees" description:"The pool fee base rate for a given pool as a percentage (0.01 to 100.00%)"`
+	MaxPriceAbsolute  float64 `long:"maxpriceabsolute" description:"The absolute maximum price to pay for a ticket (default: 100.0 Coin)"`
+	TxFee             float64 `long:"txfee" description:"Default regular tx fee per KB, for consolidations (default: 0.01 Coin/KB)"`
+	BalanceToMaintain float64 `long:"balancetomaintain" description:"Balance to try to maintain in the wallet"`
+}
+
+// loadTicketBuyerConfig initializes and parses the config using a config file.
+func loadTicketBuyerConfig() (*ticketBuyerConfig, error) {
+	loadConfigError := func(err error) (*ticketBuyerConfig, error) {
+		return nil, err
+	}
+
+	// Default config.
+	cfg := ticketBuyerConfig{
+		ConfigFile:        defaultTicketBuyerConfigFile,
+		MinPriceScale:     defaultMinPriceScale,
+		MaxPriceScale:     defaultMaxPriceScale,
+		AvgPriceMode:      defaultAvgPriceMode,
+		AvgPriceVWAPDelta: defaultAvgVWAPPriceDelta,
+		MaxFee:            defaultMaxFee,
+		MinFee:            defaultMinFee,
+		FeeSource:         defaultFeeSource,
+		MaxPerBlock:       defaultMaxPerBlock,
+		HighPricePenalty:  defaultHighPricePenalty,
+		BlocksToAvg:       defaultBlocksToAvg,
+		FeeTargetScaling:  defaultFeeTargetScaling,
+		MaxInMempool:      defaultMaxInMempool,
+		ExpiryDelta:       defaultExpiryDelta,
+
+		AccountName:      defaultAccountName,
+		MaxPriceAbsolute: defaultMaxPriceAbsolute,
+		TxFee:            defaultTxFee,
+	}
+
+	// A config file in the current directory takes precedence.
+	exists := false
+	if _, err := os.Stat(defaultTicketBuyerConfigFilename); !os.IsNotExist(err) {
```

Comment:
I'd remove all this. We don't want stuff in the current directory ever taking precedence.  Also doesn't seem to any effect on behavior since if it does exist, it just sets the config file option to the same as the default.

If the config file is missing just return the default config.

```
if _, err := os.Stat(fileName); os.IsNotExist(err) {
    return cfg, nil
}
```

I also think this should occur *after* the main program's config parsing, so if an alternate appdata dir was set, the config will be looked there and not using the default appdata setting.  Please pass that value in as a parameter here and make the default relative to that directory.